src/lib/x509/x509_ext.cpp

```diff
@@ -868,6 +877,582 @@ const std::string& TNAuthList::Entry::telephone_number() const {
    return std::get<ASN1_String>(m_data).value();
 }
 
+std::vector<uint8_t> IPAddressBlocks::encode_inner() const {
+   std::vector<uint8_t> output;
+   DER_Encoder(output).start_sequence().encode_list(m_ip_addr_blocks).end_cons();
+   return output;
+}
+
+void IPAddressBlocks::decode_inner(const std::vector<uint8_t>& in) {
+   BER_Decoder(in).decode_list(m_ip_addr_blocks).verify_end();
+}
+
+void IPAddressBlocks::sort_and_merge() {
```

Comment:
It would be helpful, both for me right now, and any future readers, if there was a comment explaining the intended behavior. Ideally with references to the RFC, if relevant

Reply:
I'll try and get a more complete answer / documentation done somewhere around Tuesday, but the tl;dr for all the merge_* functions is that they are in response to https://datatracker.ietf.org/doc/html/rfc3779#section-2.2.3.3 and the last sentences of https://datatracker.ietf.org/doc/html/rfc3779#section-2.2.3.6 (https://datatracker.ietf.org/doc/html/rfc3779#section-3.2.3.4 for the AS nums)
Essentially, we want to compress the certificate to as few values as we can, merging ranges when possible and the like, both for size reasons and to make it "easier" for users of the cert to verify

For all the other areas I'll add it to the code when I have the time soon-ish